sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs

```diff
@@ -54,9 +54,13 @@ public virtual void GlobalTimeoutTearDown()
             var timeout = TestEnvironment.GlobalIsRunningInCI ? GLOBAL_TEST_TIMEOUT_IN_SECONDS : GLOBAL_LOCAL_TEST_TIMEOUT_IN_SECONDS;
             if (duration > TimeSpan.FromSeconds(timeout))
             {
-                executionContext.CurrentResult.SetResult(
-                    ResultState.Failure,
-                    $"Test exceeded global time limit of {timeout} seconds. Duration: {duration}");
+                string message = $"Test exceeded global time limit of {timeout} seconds. Duration: {duration} ";
+                if (this is RecordedTestBase &&
+                    !executionContext.CurrentTest.GetCustomAttributes<RecordedTestAttribute>(true).Any())
+                {
+                    message += Environment.NewLine + "Add the [RecordedTest] attribute to your test to allow an automatic retry for timeouts.";
+                }
+                throw new TestTimeoutException(message);
```

Comment:
Should we still be calling `executionContext.CurrentResult.SetResult(` ? I'm not sure if a test with exception handling would potentially still pass here.

Reply:
I don't think it's necessary - this exception is raised during TearDown and I don't believe there is a way for a test to prevent this from throwing.